src/Take.Elephant.Sql/DbConnectionDecorator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Take.Elephant.Sql
+{
+    public sealed class DbConnectionDecorator : DbConnection
```

Comment:
The class name needs a more specific name, like ExpirationDbConnection.

And it will be better putting this class inside ExpirableKeySqlMap, since it is supposed to be used only there, and only in conjunction with ExpirationDatabaseDriver - those 2 decorators are an internal design detail, not supposed to be publicly available. 

Another reason to put them 'close' is that both must use the same parameter name, and both should use a single constant value for this. 

In summary, I suggest:
-Rename this class to ExpirationDbConnection
-Move it inside ExpirableKeySqlMap
-Make this class and ExpirationDatabaseDriver `internal` (instead of public or private)
-Use attribute InternalsVisibleTo to still allow unit test project to access this class (to keep things simple)

Reply:
Thanks for the sugestions , I implemented them all.